pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/codestyle/DuplicateImportsRule.java

```diff
@@ -88,22 +88,31 @@ private boolean isDisambiguationImport(ASTCompilationUnit node, String singleTyp
     public Object visit(ASTImportDeclaration node, Object data) {
         ImportWrapper wrapper = new ImportWrapper(node.getImportedName(), node.getImportedName(),
                 node, node.isStatic() && node.isImportOnDemand());
-
-        // blahhhh... this really wants to be ASTImportDeclaration to be
-        // polymorphic...
+        boolean isViolation = false;
         if (node.isImportOnDemand()) {
             if (importOnDemandImports.contains(wrapper)) {
-                addViolation(data, node, node.getImportedName());
+                for (ImportWrapper thisImportOnDemand : importOnDemandImports) {
```

Comment:
Maybe we don't need the changes here at all, if we correct the equals implementation of ImportWrapper.
A Unit/Rule test should tell us...